File: microservices/svc/echo/handler/echo.py
from echo.v1.echo_grpc import EchoBase
from echo.v1.echo_pb2 import EchoRequest, EchoResponse
from grpclib.server import Stream


class Echo(EchoBase):
    async def Echo(self, stream: Stream[EchoRequest, EchoResponse]) -> None:
        req = await stream.recv_message()
        assert req is not None

        res = EchoResponse(msg=req.msg)
        await stream.send_message(res)

File: microservices/svc/echo/handler/echo.go
package handler

import (
	"context"

	pb "github.com/kzmake/skeleton/gen/go/echo/v1"
)

type echo struct {
	pb.UnimplementedEchoServer
}

var _ pb.EchoServer = new(echo)

func NewEcho() pb.EchoServer { return &echo{} }

func (h *echo) Echo(ctx context.Context, req *pb.EchoRequest) (*pb.EchoResponse, error) {
	return &pb.EchoResponse{Msg: req.Msg}, nil
}
